src/UnityExtension/Assets/Editor/GitHub.Unity/UI/LfsLocksModificationProcessor.cs

```diff
@@ -0,0 +1,107 @@
+using System.Collections.Generic;
+using System.Linq;
+using GitHub.Logging;
+using UnityEditor;
+
+namespace GitHub.Unity
+{
+    class LfsLocksModificationProcessor : UnityEditor.AssetModificationProcessor
+    {
+        private static ILogging Logger = LogHelper.GetLogger<LfsLocksModificationProcessor>();
+        private static IRepository repository;
+        private static IPlatform platform;
+        private static List<GitLock> locks = new List<GitLock>();
+        private static CacheUpdateEvent lastLocksChangedEvent;
+
+        public static void Initialize(IRepository repo, IPlatform plat)
+        {
+            //Logger.Trace("Initialize HasRepository:{0}", repo != null);
+            repository = repo;
+            platform = plat;
+
+            if (repository != null)
+            {
+                repository.LocksChanged += RepositoryOnLocksChanged;
+                repository.CheckLocksChangedEvent(lastLocksChangedEvent);
+            }
+        }
+
+        public static string[] OnWillSaveAssets(string[] paths)
+        {
+            //Logger.Trace("OnWillSaveAssets: [{0}]", string.Join(", ", paths));
+            return paths;
+        }
+
+        public static AssetMoveResult OnWillMoveAsset(string oldPath, string newPath)
+        {
+            //Logger.Trace("OnWillMoveAsset:{0}->{1}", oldPath, newPath);
+
+            var result = AssetMoveResult.DidNotMove;
+            if (IsLocked(oldPath))
+            {
+                result = AssetMoveResult.FailedMove;
+            }
+            else if (IsLocked(newPath))
+            {
+                result = AssetMoveResult.FailedMove;
+            }
+            return result;
+        }
+
+        public static AssetDeleteResult OnWillDeleteAsset(string assetPath, RemoveAssetOptions option)
+        {
+            //Logger.Trace("OnWillDeleteAsset:{0}", assetPath);
+            return IsLocked(assetPath) ? AssetDeleteResult.FailedDelete : AssetDeleteResult.DidNotDelete;
+        }
+
+        public static bool IsOpenForEdit(string assetPath, out string message)
+        {
+            //Logger.Trace("IsOpenForEdit:{0}", assetPath);
+            var lck = GetLock(assetPath);
+            if (lck.HasValue)
+            {
+                message = "File is locked for editing by " + lck.Value.User;
+                return false;
+            }
+            else
+            {
+                message = null;
+                return true;
+            }
+        }
+
+        private static void RepositoryOnLocksChanged(CacheUpdateEvent cacheUpdateEvent)
+        {
+            if (!lastLocksChangedEvent.Equals(cacheUpdateEvent))
+            {
+                lastLocksChangedEvent = cacheUpdateEvent;
+                locks = repository.CurrentLocks;
```

Comment:
From what I've seen and read about `AssetModificationProcessor` the methods inside get called very often.
Should we create a dictionary from this list object to mitigate that?

Reply:
Yeah, probably a good idea to preformat the data here to make the lookups as fast as possible in the callbacks.